### PR TITLE
feat(cli): Not require resource name when adding

### PR DESCRIPTION
**Pull request by raulb**
# Description of change

This pull-requests changes more than I anticipated at first, but it sets a precedent on how I'd like all commands should be tested. This involves separating the logic of each command execution on their own method so they could be tested separately without necessarily testing the CLI framework (cobra in this case). There's only "cobra" thing related which is an annotation that's added to flags when they're set to required.

This starts using structs for a command so we could stop the use of global variables when they're necessary.

It also changes the output when adding a resource since we were including the name before and now that could be generated by the API.

Fixes https://meroxa.atlassian.net/browse/PLATFORM-185

Kudos to @lovromazgon for all his comments and recommendations while I've worked on this pull-request.

# Type of change

- [x]  New feature
- [ ]  Bug fix
- [x]  Refactor
- [ ]  Documentation

# Demo

**Before**

```
$ meroxa add resource --url "url" --type postgres
Error: requires resource name

Usage:
  meroxa add resource <resource-name> [flags]

$ meroxa add resource resource-name --url "url" --type postgres
Adding postgres resource (resource-name)...
Resource resource-name successfully added!
```

**Now**

```
$ meroxa add resource --url "url" --type postgres
Adding postgres resource...
postgres resource with name resource-729906 successfully added!

$ meroxa list resources
 ID        NAME           TYPE                                           URL
==== ================= ========== =================================================================================
  2   resource-729906   postgres   url

```

# How was this tested?

- [x]  Unit Tests
- [ ]  Deployed to staging

### CODE REVIEW

File: cmd/add_resource.go [hunk 1]
```diff
@@ -19,17 +19,84 @@ package cmd
...
-	"errors"
 	"fmt"
-
 	"github.com/meroxa/cli/utils"
 	"github.com/meroxa/meroxa-go"
 	"github.com/spf13/cobra"
 )
 
+type AddResourceClient interface {
+	CreateResource(ctx context.Context, resource *meroxa.CreateResourceInput) (*meroxa.Resource, error)
```
[the remark this answers is not in the document]
Comment (raulb) [reply cmd/add_resource.go:29]: @dianadoherty it's needed for mocking:

- https://github.com/meroxa/cli/pull/98/files#diff-5cb4d28640099a9ef8580667151d5d0c8aa629adb91948ad8804349a1ebbacd7R90
- https://github.com/meroxa/cli/pull/98/files#diff-4ea003c611b21fc10d1e10fdde9d51af3172a68adb48ef8f3752bcf5afd10c1dR48-R51

[the remark this answers is not in the document]
Comment (raulb) [reply cmd/add_resource.go:29]: @dianadoherty I like the idea of having a separate interface for all cli cmds. I'll definitely work on that in a separate PR!

**Review by lovromazgon — APPROVED** [review]
Nice to see a tested command :) I'd appreciate it if the codebase would be formatted with `gofmt`, otherwise looks good.

**Review by dianadoherty — APPROVED** [review]

**Comment by raulb** [comment]
@lovromazgon right! I thought I had it enabled it on my IDE. I'll make sure I lint on that next time https://github.com/meroxa/cli/pull/98/commits/107e242a67704069daa80ed86c0d21468433b6a6.

**Pull request merged**